internal/protocol/util.go

```diff
@@ -25,30 +25,33 @@
 package protocol
 
 import (
-	"fmt"
+	"errors"
 	"strings"
 
-	"github.com/gogo/protobuf/proto"
-	"github.com/gogo/protobuf/types"
 	protocolpb "go.temporal.io/api/protocol/v1"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 )
 
+var ErrProtoNameNotFound = errors.New("protocol name not found")
```

Comment:
I'm a fan of them as you can use them with `errors.Is`, and I prefer that to string matching.

I'm happy to yield to our conventions however

Reply:
We don't really have a convention here, so it's fine. I'm a fan of them _when_ I need to use them with `errors.Is`, but not for every error in the SDK just because some future user _may_ need to use them with `errors.Is`. By that logic, we'd never have anymore inline errors which I think is a bit too hardcore of a stance to take here (though may have value for other projects).

(again, no need to change)